pages/_app.js

```diff
@@ -4,12 +4,41 @@ import initialFlashcards from "@/assets/flashcards.json";
 import collections from "@/assets/collections.json";
 import useLocalStorageState from "use-local-storage-state";
 import { uid } from "uid";
+import { useState } from "react";
 
 export default function App({ Component, pageProps }) {
   const [flashcards, setFlashcards] = useLocalStorageState("flashcards", {
     defaultValue: initialFlashcards,
   });
 
+  const [currentFlashcard, setCurrentFlashcard] = useState(null);
+
+  const [actionMode, setActionMode] = useState("default");
+
+  function changeCurrentFlashcard(flashcard) {
+    setCurrentFlashcard(flashcard);
+  }
+
+  function changeActionMode(mode) {
+    setActionMode(mode);
+  }
+
+  function handleEditFlashcard(newFlashcard) {
```

Comment:
in my opinion it's crucial to add a validation check to ensure currentFlashcard isn’t null before proceeding. it is only for example: 
```suggestion
  function handleEditFlashcard(newFlashcard) {
  if (!currentFlashcard) {
    console.error("No flashcard selected for editing.");
    return;
  }
```

Reply:
Yes, not a bad idea. Although, currently, there is no situation, where currentFlashcard could be null. It's not possible. But to include error handling makes always sense.